lib/block-supports/typography.php

```diff
@@ -206,6 +206,83 @@ function gutenberg_typography_get_css_variable_inline_style( $attributes, $featu
 	return sprintf( '%s:var(--wp--preset--%s--%s);', $css_property, $css_property, $slug );
 }
 
+/**
+ * Returns a font-size value based on a given font-size preset. If typography.fluid is enabled it will calculate clamp values.
+ *
+ * @param array $preset Duotone preset value as seen in theme.json.
+ * @return string        Font-size value.
+ */
+function gutenberg_get_typography_font_size_value( $preset ) {
+	$typography_settings = gutenberg_get_global_settings( array( 'typography' ) );
+
+	// This is where we'll keep options I guess.
+	if ( ! isset( $typography_settings['fluid'] ) ) {
+		return $preset['size'];
+	}
+
+	// Up for discussion.
+	$default_unit                   = 'rem';
+	$default_minimum_viewport_width = '1600px';
+	$default_maximum_viewport_width = '650px';
+
+	// Matches rem or px values only.
```

Comment:
Do we even need to support more units? Maybe `em`

Reply:
> I think adding em would be nice.

We could allow `em` out of the box. I'm assuming most designers will recognize that the resulting font-size values will be relative to the parent element's font size.

There's also the question of having consistent units and values for the purposes of calculating the clamp values.

I've thrown a kitchen sink at this experiment - the calculations rely on `px` or `rem` units (the latter calculated by dividing `px_value/16`. 

There's probably scope to simplify things a great deal.

Reply:
> the calculations rely on px or rem units (the latter calculated by dividing px_value/16

In many cases the root font-size can be something other than 16px... So this calculation will not be correct 100% of the time, but should cover a lot of scenarios 👍

Reply:
> In many cases the root font-size can be something other than 16px.

Very good point 😄 I'll see how we can make this more flexible. Thanks a lot!